src/librustc_mir/borrow_check.rs

```diff
@@ -320,6 +323,7 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> DataflowResultsConsumer<'b, 'tcx>
                                    (drop_lvalue, span),
                                    Deep,
                                    JustWrite,
+                                   LocalMutationIsAllowed::No,
```

Comment:
`dropandreplace` is an assignment of a type with a destructor, so this should be an error only if the value is already initialized.

e.g. the string example:
```Rust
let s = String::new();
s = String::new();
```